openstack/compute/v2/availabilityzones/utils.go

```diff
@@ -1,27 +1,34 @@
 package availabilityzones
 
 import (
+	"fmt"
+
 	"github.com/gophercloud/gophercloud"
 	"github.com/gophercloud/gophercloud/openstack/compute/v2/extensions/availabilityzones"
 )
 
 // ListAvailableAvailabilityZones is a convenience function that return a slice of available Availability Zones.
 func ListAvailableAvailabilityZones(client *gophercloud.ServiceClient) ([]string, error) {
-	var ret []string
+	var zones []string
+
 	allPages, err := availabilityzones.List(client).AllPages()
 	if err != nil {
-		return ret, err
+		return zones, err
 	}
 
 	availabilityZoneInfo, err := availabilityzones.ExtractAvailabilityZones(allPages)
 	if err != nil {
-		return ret, err
+		return zones, err
+	}
+
+	if len(availabilityZoneInfo) == 0 {
+		return nil, fmt.Errorf("could not find an available compute availability zone")
```

Comment:
Why are you returning an error when there are zero availability zones? I tend to reserve a non-nil error value for failures. Do you think the API would not legitimately return zero AZs?

Reply:
Yeah, this deserves a comment. Nova will always return at least two AZs: one for internal services and one for everything else. These are typically called `internal` and `nova` but you can configure them with `[availability_zone] internal_service_availability_zone` and `[availability_zone] default_availability_zone`,

Reply:
Perhaps this is unnecessary logic though and I could leave this to the caller to acknowledge. wdyt?

Reply:
Thanks for the explanation.
It sounds fair to me to let the user deal with the actual Nova response, rather than trying to fix it by raising an error.

Reply:
Done